add shared libraries for C++

**Image I'm using:**
v0.46.0


**Issue or Feature Request:**
Without a shared `libstdc++.so`, attempts to use the static `libstdc++.a` can fail if shared linking is required:
```
 #14 144.2 /usr/lib/gcc/x86_64-bottlerocket-linux-gnu/11/../../../../x86_64-bottlerocket-linux-gnu/bin/ld: /x86_64-bottlerocket-linux-gnu/sys-root/usr/lib/libstdc++.a(eh_globals.o): relocation R_X86_64_TPOFF32 against `_ZZN12_GLOBAL__N_110get_globalEvE6global' can not be used when making a shared object; recompile with -fPIC
  #14 144.2 /usr/lib/gcc/x86_64-bottlerocket-linux-gnu/11/../../../../x86_64-bottlerocket-linux-gnu/bin/ld: failed to set dynamic section sizes: bad value
  #14 144.2 collect2: error: ld returned 1 exit status
```

This prevents building the NVIDIA DCGM exporter, which is written in C++ and relies on shared libraries loaded as plugins.

--- a/patches/buildroot/0004-build-gcc-with-TOOLS_DIR-and-SYSROOT_DIR.patch
+++ b/patches/buildroot/0004-build-gcc-with-TOOLS_DIR-and-SYSROOT_DIR.patch
@@ -84,7 +84,7 @@ index aa883beb7b..0f61ff0df1 100644
 -else
 -HOST_GCC_FINAL_CONF_OPTS += --enable-shared
 +ifeq (y,y)
-+HOST_GCC_FINAL_CONF_OPTS += --enable-shared=libgcc
++HOST_GCC_FINAL_CONF_OPTS += --enable-shared=libgcc,libstdc++
  endif
  
  ifeq ($(BR2_GCC_ENABLE_OPENMP),y)

